### DISCUSSION TITLE
How to add an alternative option to formula (or create cast) that compiles program with GUI support?

### DISCUSSION THREAD

**Discussion by munro** [Homebrew/discussions#5444] [Writing Formulae/Casks]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

hey all, I want to add the GUI version of SWI-Prolog to homebrew, but I have some barriers I'm not sure how to overcome.

The official GUI built version for macOS is unsigned, so I can't add it as a cask[^1], at least not until it's signed[^2], if the devs will even do that.

I still would love to add it, because the `gtrace` is really nice, and so is homebrew.  The core package essentially just needs to have the GUI build flags turned on[^3], and deps added[^4].

So I'm kind of wondering what to do, thoughts lingering:
- should it be cask?  thinking no, from what I can tell, casks cannot be "built" like formulas.  Me personally, I don't need the `/Applications/SWI-Prolog.app`, I just need `gtrace.` to work in the terminal `swipl` program.
- should this be built as an alternative formula?  I see in docs Homebrew/core cannot have `:optional` dependencies.
- should I it just have `_X=on` by default?  not sure if that's acceptable for everyone though...
- it requires xquartz, so if it's a formula, it can't depend on a cask?
- can I build with X enabled using only formulas?  I'm not sure how the x deps work...  but if that was the case then I can keep it as a formula, and caveat that the xquartz cast needs to be installed so `gtrace` works (this seems the most ideal).  But not sure what build packages are need to compile with X extensions enabled.

I know I can create my own tap, but I was hoping there's some solution using the official swi-prolog repos/downloads to include into offical homebrew core/cask, so that others can use and it with more trust.

any help/thoughts/support is appreciated!  thanks!

[^1]: https://github.com/Homebrew/homebrew-cask/pull/176803
[^2]: https://github.com/SWI-Prolog/swipl-devel/issues/1288
[^3]: `-DSWIPL_PACKAGES_X=ON` https://github.com/Homebrew/homebrew-core/blob/9da60e3fc1c4de3160e8695ac32be3321b324506/Formula/s/swi-prolog.rb#L50
[^4]: I believe `freetype, libx11, :cask xquartz`

**Comment by SMillerDev** [comment]
This sounds like it would massively increase the dependency tree, so it doesn't seem like something the Homebrew/core formula should do. You can easily maintain your own cask or a formula for it in your own tap like you mentioned. In the end the only person who needs to trust it is you, since I've never seen a request like this before.
